cs/src/Connections/TunnelHostBase.cs

```diff
@@ -183,22 +192,57 @@ public async Task StartAsync(Tunnel tunnel, CancellationToken cancellation)
 
             // When forwarding from a Remote port we assume that the RemotePortNumber
             // and requested LocalPortNumber are the same.
+            var portNumber = (int)port.PortNumber;
+
+            if (pfs.LocalForwardedPorts.Any((p) => p.LocalPort == portNumber || p.RemotePort == portNumber))
```

Comment:
I think `p.LocalPort` could be different if the requested port number was in use. Anyway wouldn't it be sufficient to just compare to `p.RemotePort`?

Reply:
It needs to check `p.LocalPort` only, not `p.RemotePort`, which may be random if the forwarded port is used on the client machine.